tests/test_compiler.py

```diff
@@ -605,3 +607,52 @@ def test_caching_dependency_kfp(tmp_path_factory):
         second_component_cache_key_dict[arg_list[0]]
         != second_component_cache_key_dict[arg_list[1]]
     )
+
+
+def test_sagemaker_build_command():
+    compiler = SagemakerCompiler()
+    metadata = Metadata(
+        pipeline_name="example_pipeline",
+        base_path="/foo/bar",
+        component_id="component_2",
+        run_id="example_pipeline_2024",
+        cache_key="42",
+    )
+    args = {"foo": "bar", "baz": "qux"}
+    command = compiler.build_command(metadata, args)
+
+    assert command == [
+        "--metadata",
+        '\'{"base_path": "/foo/bar", "pipeline_name": "example_pipeline", '
+        '"run_id": "example_pipeline_2024", "component_id": "component_2", '
+        '"cache_key": "42"}\'',
+        "--output_manifest_path",
+        "/foo/bar/example_pipeline/example_pipeline_2024/component_2/manifest.json",
+        "--foo",
+        "'bar'",
+        "--baz",
+        "'qux'",
+    ]
+
+    # with dependencies
+    dependencies = ["component_2"]
```

Comment:
small nitpick but would name it `component_1`. Right now it looks like the component is dependent on itself